odoo/addons/base/tests/test_i18n.py

```diff
@@ -0,0 +1,32 @@
+from odoo.tests import TransactionCase
+from odoo.tools.i18n import format_list
+
+
+class I18nTest(TransactionCase):
+    def test_format_list(self):
+        lang = self.env["res.lang"]
+
+        formatted_text = format_list(self.env, ["Mario", "Luigi"])
+        self.assertEqual(formatted_text, "Mario and Luigi", "Should default to English.")
+
+        formatted_text = format_list(self.env, ["To be", "Not to be"], "or")
+        self.assertEqual(formatted_text, "To be or Not to be", "Should take the style into account.")
+
+        lang._activate_lang("fr_FR")
+
+        formatted_text = format_list(lang.with_context(lang="fr_FR").env, ["Athos", "Porthos", "Aramis"])
+        self.assertEqual(formatted_text, "Athos, Porthos et Aramis", "Should use the language of the user.")
+
+        formatted_text = format_list(
+            lang.with_context(lang="en_US").env, ["Athos", "Porthos", "Aramis"], lang_code="fr_FR"
+        )
+        self.assertEqual(formatted_text, "Athos, Porthos et Aramis", "Should use the chosen language.")
+
+        formatted_text = format_list(
+            lang.with_context(lang="fr_FR").env, ["Athos", "Porthos", "Aramis"], lang_code="es_ES"
+        )
+        self.assertEqual(
+            formatted_text,
+            "Athos, Porthos et Aramis",
+            "Should use the language of the user if the chosen language is not installed.",
+        )
```

Comment:
What's the reason for not always using the provided language?

Reply:
I guess the question is more "Why would you want to specify a language rather than use the one of the user?"
I left the parameter from the babel function because it might have some niche use case that I couldn't think of, but it wouldn't make sense to display stuff in a different language than what the rest of the page is displayed in, even more so if said language is not installed.

Reply:
I understand the weirdness of displaying something in a language that is not the user's language, but since you provide a way to do this for the sake of "the niche use case we can't think of" anyway, I don't see any reason to limit it to the installed languages either. The default behavior is what developers want 99% of the time. If they bother to explicitly provide a language, they must have their reasons, I don't want to prevent them from doing so. I find it even worse if they explicitly provide a language and it ends up not being used for some untransparent reason (i.e. the language is not installed)

Reply:
Fixed